pkg/sources/gitlab/gitlab.go

```diff
@@ -315,28 +319,28 @@ func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk)
 			}
 			defer os.RemoveAll(path)
 			if err != nil {
-				errsMut.Lock()
-				errs = append(errs, err)
-				errsMut.Unlock()
-				return
+				scanErrs = append(scanErrs, err)
+				return nil
 			}
```

Comment:
I think we might still need to lock here if we are concurrently appending to `scanErrs`.

Reply:
Random thought - a named return parameter / deferred function might work well here:

```go

s.jobPool.Go(func() (err error) {
    defer func() {
        if err != nil {
            // lock and append to scanErrs
         }
         // always set err to nil to prevent marking as error
         err = nil
    }()
})